synapse/util/macaroons.py

```diff
@@ -16,14 +16,18 @@
 
 """Utilities for manipulating macaroons"""
 
-from typing import Callable
+from typing import Callable, Optional
 
 import pymacaroons
+from pymacaroons.exceptions import MacaroonVerificationFailedException
 
 
 def get_value_from_macaroon(macaroon: pymacaroons.Macaroon, key: str) -> str:
     """Extracts a caveat value from a macaroon token.
 
+    Checks that there is exactly one caveat of the form "key = <val>" in the macaroon,
+    and returns the extracted value.
+
```

Comment:
This change is being introduced mostly in the interests of safety. We've been burnt badly by this in the past (https://matrix.org/blog/2016/07/08/critical-security-vulnerability-in-synapse-0-12-to-0-16-1-inclusive), so some double-checking feels like it is in order.